src/compile.rs

```diff
@@ -332,6 +332,11 @@ pub fn validate_functions<'a>(
             Ok(_) => (),
             Err(e) => errors.append(e),
         }
+    }
+
+    derive_functions(&mut functions, types, class_perms)?;
```

Comment:
What this `fixup! Implement the core of the @derive annotation` commit fix? Could you add a test to prevent this kind of future regression?

Reply:
This is tested by the test added in a3247c0abd2c6c9e37687cda9e6848fa821722d6.  Adding that test exposed the bug.  I intended to squash the fixup commit into the original prior to pushing, but forgot, so it will be squashed on the next push.